### PR TITLE
MM-18508 For empty or broken links - show a broken image

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
Added image load validation as well as 'empty src' check to display a broken image

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-18508

### CODE REVIEW

File: components/markdown_image.jsx [hunk 1]
```diff
@@ -43,9 +46,19 @@ export default class MarkdownImage extends React.PureComponent {
...
+        this.setState({loadFailed: true});
+    }
+
     render() {
         const {imageMetadata, src, imageIsLink} = this.props;
-
+        if (src === '' || this.state.loadFailed) {
+            return (
+                <img
+                    alt='Broken link'
```
Comment (hmhealey) [inline components/markdown_image.jsx]: I think this should use `this.props.alt` instead

File: components/markdown_image.jsx [hunk 2]
```diff
@@ -43,9 +46,19 @@ export default class MarkdownImage extends React.PureComponent {
...
+                    src={brokenImageIcon}
+                />);
```
Comment (hmhealey) [inline components/markdown_image.jsx]: Minor nit, can you move the closing bracket onto its own line?

**Comment by lindy65** [comment]
Thanks @reflog,

This looks good to me, I see a broken image when posting a broken or empty link.

**Comment by reflog** [comment]
@lindy65 @amyblais this PR depends on another one: https://github.com/mattermost/mattermost-webapp/pull/3639 which is slated for 5.18 .
Should we move this one to 5.18 or cherrypick https://github.com/mattermost/mattermost-webapp/pull/3639 into 5.17 as well?

**Comment by lindy65** [comment]
Thanks @reflog 

#3639 is a feature story which shouldn't go into v5.17 (being a maintenance release) so I'd be inclined to move this PR to v5.18 but I'll defer to Amy for a final decision.

**Pull request merged**